packages/theming/src/utils/StyledBaseIcon.ts

```diff
@@ -5,12 +5,15 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import styled from 'styled-components';
-import React, { Children } from 'react';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const StyledBaseIcon = styled(({ children, theme, ...props }) =>
-  React.cloneElement(Children.only(children), props)
+import styled, { DefaultTheme } from 'styled-components';
+import { Children, cloneElement, PropsWithChildren, ReactElement, SVGAttributes } from 'react';
+export interface IStyledBaseIconProps extends PropsWithChildren<SVGAttributes<SVGElement>> {
+  theme?: DefaultTheme;
+}
+export const StyledBaseIcon = styled(
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  ({ children, theme, ...props }: IStyledBaseIconProps) =>
+    cloneElement(Children.only(children as ReactElement), props)
```

Comment:
```suggestion
import styled, { DefaultTheme } from 'styled-components';
import { Children, cloneElement, PropsWithChildren, ReactElement, SVGAttributes } from 'react';

export interface IStyledBaseIconProps extends PropsWithChildren<SVGAttributes<SVGElement>> {
  theme?: DefaultTheme;
}

export const StyledBaseIcon = styled(
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ({ children, theme, ...props }: IStyledBaseIconProps) =>
    cloneElement(Children.only(children as ReactElement), props)
```

spacing plz :pray: for improved readability